internal/providersdkv2/resource_vault_cluster.go

```diff
@@ -1191,6 +1332,37 @@ func getObservabilityConfig(propertyName string, d *schema.ResourceData) (*vault
 	return getValidObservabilityConfig(config)
 }
 
+// if http observability information is provided, this function ensures that authentication fields are valid and returns the authentication method used
+func validateHTTPAuth(httpBasicUser string, httpBasicPassword string, httpBearerToken string) (diag.Diagnostics, *vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth, *vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth) {
+	var httpConfigError diag.Diagnostics
+
+	// only one of basic or bearer authentication should be submitted
+	if httpBearerToken != "" && httpBasicUser != "" || httpBearerToken != "" && httpBasicPassword != "" {
```

Comment:
can't this expression be simplified?
```suggestion
	if httpBearerToken != "" && (httpBasicUser != "" || httpBasicPassword != "") {
```

Reply:
you need the parenthesis otherwise the && operator takes precedence